core/client/src/main/java/alluxio/hadoop/HdfsFileInputStream.java

```diff
@@ -105,49 +85,18 @@ public int available() throws IOException {
     if (mClosed) {
       throw new IOException("Cannot query available bytes from a closed stream.");
     }
-    return (int) mAlluxioFileInputStream.remaining();
+    return (int) mInputStream.remaining();
   }
 
   @Override
   public void close() throws IOException {
-    if (mAlluxioFileInputStream != null) {
-      mAlluxioFileInputStream.close();
-    }
-    if (mHdfsInputStream != null) {
-      mHdfsInputStream.close();
+    if (mClosed) {
+      return;
     }
+    mInputStream.close();
     mClosed = true;
   }
 
-  /**
-   * Sets {@link #mHdfsInputStream} to a stream from the under storage system with the stream
-   * starting at {@link #mCurrentPosition}.
-   *
-   * @throws IOException if opening the file fails
-   */
-  // TODO(calvin): Consider removing this when the recovery logic is available in FileInStream
```

Comment:
Good point, I think we should remove this now regardless of if there is recovery logic, since the behavior will be equivalent to if the user was using the native file system interface. I've also opened a ticket to track this TODO since it is no longer in our codebase: https://alluxio.atlassian.net/browse/ALLUXIO-2742